Exame_2017/prob2/prob2.c

```diff
@@ -8,15 +8,69 @@
 /*** problema 2.1 ***/
 grafo * grafo_intersecao(grafo *g1, grafo *g2)
 {
+	if(g1 == NULL){
+		return NULL;
+	}
+
+	if(g2 == NULL){
+		return NULL;
+	}
+
+	if(g1->tamanho != g2->tamanho){
+		return NULL;
+	}
 
-	return NULL;
+	grafo * g = grafo_novo(g1->tamanho, g1->tipo);
+
+	if(g == NULL){
+		return NULL;
+	}
+
+	lista_no *no;
+
+	for(int i=0;i<g1->tamanho;i++)
+	{
+		no = g1->adjacencias[i].inicio;
+		while(no != NULL) {
+			if(grafo_aresta(g2,i,no->vertice)){
+				grafo_adiciona(g,i,no->vertice);
+			}
+			no = no->proximo;
+		}
+	}
+	
+	return g;
 }
 
 /*** problema 2.2 ***/
 void imprime_ordenado(tabela_dispersao *td)
 {
-	/* complexidade do algoritmo: ________________ */
+	/* complexidade do algoritmo: O(n log n)*/
+	if(td == NULL){
+		return;
+	}
+	
+	int n;
+	objeto *obj;
+	heap *h;
+
+	h = heap_nova(100);
```

Comment:
what is 100?

Reply:
Random number big enough, it doesn't have meaning.
What if I change it to ` h = heap_nova(tabela_numelementos(td));` ?